runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,311 @@
+			deathTimer.setTooltip("Died on world: " + config.deathWorld());
+			infoBoxManager.addInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() == ChatMessageType.SERVER)
+		{
+			if (event.getMessage().equals("Oh dear, you are dead!"))
+			{
+				config.hasRespawned(true);
+			}
+
+			if ((!config.hasDied()) && (config.timeOfDeath().equals("")))
+			{
+				return;
+			}
+
+			if (event.getMessage().equals("Welcome to RuneScape.") && config.showDeathInfoBox())
+			{
+				if (Instant.now().isBefore(Instant.parse(config.timeOfDeath()).plus(1, ChronoUnit.HOURS)))
+				{
+					if (deathTimer != null)
+					{
+						infoBoxManager.removeInfoBox(deathTimer);
+						deathTimer = null;
+					}
+					deathTimer = new Timer(Duration.ofHours(1).minus(Duration.between(Instant.parse(config.timeOfDeath()), Instant.now())).getSeconds(), ChronoUnit.SECONDS, BONES, this);
+					deathTimer.setTooltip("Died on world: " + config.deathWorld());
+					infoBoxManager.addInfoBox(deathTimer);
+				}
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!config.hasDied() || !config.hasRespawned() || (client.getWorld() != config.deathWorld()))
+		{
+			return;
+		}
+
+		if ((client.getLocalPlayer().getWorldLocation().getX() == config.deathLocationX())
+			&& (client.getLocalPlayer().getWorldLocation().getY() == config.deathLocationY())
+			&& (client.getLocalPlayer().getWorldLocation().getPlane() == config.deathLocationPlane())
+			|| ((deathTimer != null) && (deathTimer.getText().equals("0:00"))))
+		{
+			if (client.hasHintArrow())
+			{
+				client.clearHintArrow();
+			}
+
+			if (deathTimer != null)
+			{
+				infoBoxManager.removeInfoBox(deathTimer);
+				deathTimer = null;
+			}
+
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			config.deathMarkerReset(true);
+		}
+
+		if (config.deathMarkerReset())
+		{
+			config.hasDied(false);
+			config.hasRespawned(true);
+			config.deathMarkerReset(false);
+			config.deathWorld(-1);
+			config.deathLocationX(-1);
+			config.deathLocationY(-1);
+			config.deathLocationPlane(-1);
+			config.timeOfDeath("");
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("deathIndicator"))
+		{
+			if (!config.showDeathHintArrow() && client.hasHintArrow())
+			{
+				client.clearHintArrow();
+			}
+
+			if (!config.showDeathInfoBox() && deathTimer != null)
+			{
+				infoBoxManager.removeInfoBox(deathTimer);
+				deathTimer = null;
+			}
+
+			if (!config.showDeathOnWorldMap())
+			{
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			}
+
+			// Simulates hitting the reset button, removes death info!
+			if ((config.deathLocationX() == -1)
+				&& (config.deathLocationY() == -1)
+				&& (config.deathLocationPlane() == -1)
+				&& (!config.hasDied())
+				&& (config.hasRespawned())
```

Comment:
Why you need hasRespawned and hasDied? All you need is null-check timeOfDeath

Reply:
I was using hasDied as a check (for efficency purposes) to make sure the player has at least died before doing something, the hasRespawned comes into play for resetting the markers. If I don't have the hasRespawned boolean, the marker is set and then instantly cleared.